view/frontend/web/js/view/summary/surcharge.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @by SwiftOtter, Inc. 3/30/17
+ * @website https://swiftotter.com
+ **/
+define([
+    'jquery',
+    'Magento_Checkout/js/view/summary/abstract-total',
+    'Magento_Checkout/js/model/quote',
+    'Magento_Checkout/js/model/totals'
+], function($, Component, quote, totals) {
+    return Component.extend({
+        quote: quote,
+        totals: quote.getTotals(),
+
+        formattedSurchargeValue: function () {
+            return this.getFormattedPrice(this.getSurchargeValue());
+        },
+
+        hasSurcharge: function () {
+            return !!this.getSurchargeValue();
+        },
+
+        getSurchargeValue: function() {
+            var surcharge = totals.getSegment('shipping_surcharge');
+
+            if (surcharge && surcharge.value) {
+                return surcharge.value;
+            }
+
+            return 0;
```

Comment:
Again, you might want to move this into an else, as opposed to leaving it like this. This isn't bad, but it gives the impression of the early return statement halting function execution early, instead of being one of two options for leaving execution at the same time.